visualization/app/codeCharta/util/urlExtractor.ts

```diff
@@ -39,15 +40,44 @@ export class UrlExtractor {
 		if (!fileName) {
 			throw new Error(`Filename is missing`)
 		}
+		if (fileName.endsWith(".gz")) {
+			return this.getUnzippedFile(fileName)
+		}
+		return this.getFile(fileName)
+	}
 
+	private async getUnzippedFile(fileName: string): Promise<NameDataPair> {
+		const options = { responseType: "blob" as const }
+		const response = await this.$http.get<Blob>(fileName, options)
+		if (response.status >= 200 && response.status < 300) {
+			return new Promise(resolve => {
+				const reader = new FileReader()
+				reader.readAsArrayBuffer(response.data)
+				let content: ExportCCFile
+
+				reader.onload = event => {
+					const readerContent = zlib.unzipSync(Buffer.from(<string>event.target.result)).toString()
```

Comment:
Using the blocking variant is not ideal as it'll block until it's fully unzipped. Instead, use the async version as it won't block the program from running in the meanwhile.

Reply:
@BridgeAR I'm not sure what you mean. Please, could you specify it or give an example of how to do it?

Reply:
It is already resolved by using `pako` instead of `zlib`. `zlib` has blocking ("sync") and non-blocking operations (async, the regular name without sync). Sync APIs should pretty much never be used besides rare exceptions, if they exist next to an async one.